From ab0474ea0dae399dcd256aa520089450a3363f33 Mon Sep 17 00:00:00 2001
From: zhangli <zhanggaofeng9@huawei.com>
Date: Wed, 19 Jun 2019 19:38:37 +0000
Subject: [PATCH] JDK-8135318 backport

Summary: CMS: wrong max_eden_size for check_gc_overhead_limit
LLT:
Bug url: https://bugs.openjdk.java.net/browse/JDK-8135318
---
 .../concurrentMarkSweep/concurrentMarkSweepGeneration.cpp          | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/hotspot/src/share/vm/gc_implementation/concurrentMarkSweep/concurrentMarkSweepGeneration.cpp b/hotspot/src/share/vm/gc_implementation/concurrentMarkSweep/concurrentMarkSweepGeneration.cpp
index 1872e72e36..2d7791138c 100644
--- a/hotspot/src/share/vm/gc_implementation/concurrentMarkSweep/concurrentMarkSweepGeneration.cpp
+++ b/hotspot/src/share/vm/gc_implementation/concurrentMarkSweep/concurrentMarkSweepGeneration.cpp
@@ -1924,9 +1924,10 @@ NOT_PRODUCT(
 
     // Has the GC time limit been exceeded?
     DefNewGeneration* young_gen = _young_gen->as_DefNewGeneration();
-    size_t max_eden_size = young_gen->max_capacity() -
-                           young_gen->to()->capacity() -
-                           young_gen->from()->capacity();
+
+    //PR JDK-8135318 back port
+    size_t max_eden_size = young_gen->max_eden_size();
+
     GenCollectedHeap* gch = GenCollectedHeap::heap();
     GCCause::Cause gc_cause = gch->gc_cause();
     size_policy()->check_gc_overhead_limit(_young_gen->used(),
-- 
2.12.3

